pulser/simulation/simconfig.py

```diff
@@ -0,0 +1,129 @@
+
+    def __str__(self) -> str:
+        s = ""
+        s += "Options:\n"
+        s += "----------       \n"
+        s += "noise types:       " + str(self.noise) + "\n"
+        s += "spam dictionary:   " + str(self.spam_dict) + "\n"
+        s += "temperature:       " + str(self.temperature) + "K" + "\n"
+        s += "number of runs:    " + str(self.runs) + "\n"
+        s += "samples per runs:  " + str(self.samples_per_run) + "\n"
+        s += "laser waist:       " + str(self.laser_waist) + "μm \n"
+        s += "\n" + "Solver Options: \n"+str(self.solver_options)[10:-1]+"\n"
+        return s
+
+    @property
+    def noise(self) -> List[str]:
+        return self._noise
+
+    @noise.setter
+    def noise(self, noise_types: List[str]) -> None:
+        self._noise: List[str] = []
+        for noise_type in noise_types:
+            self._add_noise(noise_type)
+
+    def _add_noise(self, noise_type: str) -> None:
+        """Adds a noise model to the SimConfig instance, to be used
+            in Simulation.
+            Args:
+                noise_type (str): Choose among:
+                    'dephasing': random phase (Z) flip
+                    'doppler': Noisy doppler runs
+                    'amplitude': Noisy gaussian beam
+                    'SPAM': SPAM errors. Adds:
+                        eta: Probability of each atom to be badly prepared
+                        epsilon: false positives
+                        epsilon_prime: false negatives
+        """
+        # Check proper input:
+        if noise_type not in noise_dict_set:
+            raise ValueError('Not a valid noise type')
+        self._noise.append(noise_type)
+
+    @property
+    def spam_dict(self) -> dict[str, float]:
+        return self._spam_dict
+
+    def set_spam(self, **values: float) -> None:
+        """Allows the user to change SPAM parameters in dictionary"""
+        for param in values:
+            if param not in {'eta', 'epsilon', 'epsilon_prime'}:
+                raise ValueError('Not a valid SPAM parameter')
+            if values[param] > 1 or values[param] < 0:
+                raise ValueError('Invalid value : must be between 0 and 1')
+            self._spam_dict[param] = values[param]
+
+    def init_spam(self) -> None:
+        self._spam_dict: dict[str, float] = \
+            {'eta': 0.005, 'epsilon': 0.01, 'epsilon_prime': 0.05}
+
+    @property
+    def temperature(self) -> float:
+        return self._temperature
+
+    @temperature.setter
+    def temperature(self, value: float) -> None:
+        # value set in microkelvin
+        self._temperature = value * 10.**-6
+        self._calc_sigma_doppler()
+
+    @property
+    def doppler_sigma(self) -> float:
+        return self._doppler_sigma
+
+    def _calc_sigma_doppler(self) -> None:
+        # sigma = keff Deltav, keff = 8.7mum^-1, Deltav = sqrt(kB T / m)
+        self._doppler_sigma: float = \
+            8.7 * np.sqrt(1.38e-23 * self._temperature / 1.45e-25)
```

Comment:
nit: I usually dislike using `\`, so I suggest this instead:
```suggestion
        self._doppler_sigma: float = 8.7 * np.sqrt(
            1.38e-23 * self._temperature / 1.45e-25)
```
Maybe the indentation on the last line is wrong, see if you can fit it